### PR TITLE
[Ready for Review] Revamps Wish Granters

### PR DESCRIPTION
**Pull request by Michiyamenotehifunana**
:cl: Toriate
tweak: Wish Granters have been revamped to have a list of selectable wishes. Find out what they do at your own peril!
balance: Eutactic blades have had their numbers nerfed drastically.
/:cl:

This PR serves to address the complaints about wishgranters being free self-antag tokens. They now have a variety of different selectable wishes, with extremely selfish options giving more negative effects, while relatively harmless wishes create challenges for the user to face while also rewarding them.

Currently, there are five wishes.

The first, power, gives the user an infinite powercell while shoving a delaminating SM shard in their faces.

The second, wealth, gives the user 2 crates with 500,000 dollars in them while also turning them into a gold golem.

The third, "I want the station to disappear", teleports the user to the station while blinding them.

The fourth, To Kill, gives the user a non-eutactic blade and spawns a Venus Man Eater with 1500 health and ramped up aggressiveness that will drop a Hyper Eutactic Blade on death. It also gives the user a 0.8 attack delay multiplier.

The fifth, nothing, makes the Wish Granter disappear.

Antagonists who use the Wish Granter will gain the generic set of powers (TK. Xray, coldresist), as well as getting their attack delay halved.

This PR also nerfs the NEB and HEB to be not OP. Further number tweaking may be necessary, depending on how it pans out.

### CODE REVIEW

**Comment by CameronWoof** [comment]
This does not fix the problem. The only reason you should ever use this item is if you are an antagonist. Any crewmember using this item should be warned or banned, per the server rules. 

Make it check for antag status and dust you if you aren't an antagonist.

**Comment by ZeroNetAlpha** [comment]
Agreed. Tis 'Self-Antag, the semi-legal way.' 100% or bust.

File: modular_citadel/code/game/machinery/wishgranter.dm [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+	else
+		to_chat(user, "You speak.  [pick("I want the station to disappear","Humanity is corrupt, mankind must be destroyed","I want to be rich", "I want to rule the world","I want immortality.")].  The Wish Granter answers.")
+		if(alert("Are you really sure?", "Confirm decision", "Yes", "No") == "Yes")
+			if(alert("Are you really, really sure?", "Confirm decision", "Yes", "No") == "Yes")
+				if(alert("You have a chance of being dusted if you keep clicking yes. Are you sure?", "Confirm decision", "Yes", "No") == "Yes")
+					if(alert("Do you want to self-antag so badly you'd risk being dusted?", "Confirm decision", "Yes", "No") == "Yes")
+						if(alert("Really?", "Confirm decision", "Yes", "No") == "Yes")
+							if (prob(75))
+								to_chat(user, "Your head pounds for a moment, before your vision clears. You instantly regret all of your life choices up to this point. Your flesh rapidly flashes away into dust.")
```
Comment (ZeroNetAlpha) [inline modular_citadel/code/game/machinery/wishgranter.dm]: "Your head pounds for a moment before your vision clears. You instantly regret all of your life choices up to this point as you watch your flesh rapidly flash away into dust." Possible alteration.

**Comment by LetterJay** [comment]
Lol

File: modular_citadel/code/game/machinery/wishgranter.dm (hunk 1 above)
Comment (ZeroNetAlpha) [reply modular_citadel/code/game/machinery/wishgranter.dm]: Even better.
"You fool. You knew this was bad from the beginning. Now you have nothing."

**Comment by silicons** [comment]
no.
make it do something not so damn griefy (hijack is pretty damn griefy)
it's a wish granter not a die of fate, this isn't fun for anyone.

**Review by silicons — DISMISSED** [review]
^

**Comment by deathride58** [comment]
In STALKER, where the wishgranter got its inspiration from, the wishgranter in all of the game's endings always ended up killing the user. It acts as a monkey's paw. A wish like "I want the zone to disappear" resulted in the wisher going blind, "I wish to be rich" resulted in the wisher being drowned and crushed by riches, etc etc.

Could probably do some tweaking to make it work more as a monkey's paw, like STALKER's wishgranter, rather than the die of fate these changes make it.

**Comment by LetterJay** [comment]
Problem is when you start adding 100% ways for it to kill you, people are just going to avoid it and it'll have virtually no use.

**Comment by CameronWoof** [comment]
If a bad item gets added to the game, why are coders other than the person who made the item expected to work harder to make the item good instead of just removing it?

**Comment by Michiyamenotehifunana** [comment]
Because it's an open source project and it's always a fun challenge to do new stuff!

**Review by deathride58 — DISMISSED** [review]

**Comment by lightcat** [comment]
I'd rather not have this kill the user in every scenario.

File: modular_citadel/code/game/machinery/wishgranter.dm [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+		if(is_special_character(user))
+			to_chat(user, "You speak.  [pick("I want power","Humanity is corrupt, mankind must be destroyed", "I want to rule the world","I want immortality")].  The Wish Granter answers.")
+			to_chat(user, "Your head pounds for a moment, before your vision clears. The Wish Granter has given you limitless power, and it's all yours!")
+			user.dna.add_mutation(HULK)
+			user.dna.add_mutation(XRAY)
+			user.dna.add_mutation(COLDRES)
+			user.dna.add_mutation(TK)
+			charges--
+			insisting = FALSE
+			to_chat(user, "You have a very great feeling about this!")
```
Comment (deathride58) [inline modular_citadel/code/game/machinery/wishgranter.dm]: Tbh, thinking about it a little, you could probably make this add a 0.5 click delay multiplier to make getting the wishgranter a pure reward for antags.

File: modular_citadel/code/game/machinery/wishgranter.dm [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+					if(!charges)
+						qdel(src)
+				if("To Kill")	//Makes you kill things in exchange for rewards!
+					to_chat(user, "<B>Your wish is granted, but at a terrible cost...</B>")
+					to_chat(user, "The Wish Granter punishes you for your wickedness, warping itself into a dastardly creature for you to kill! ...but it almost seems to reward you for this.")
+					var/obj/item/melee/transforming/energy/sword/cx/killreward = new /obj/item/melee/transforming/energy/sword/cx(get_turf(user))
+					if(user.put_in_hands(killreward))
+						to_chat(user, "[killreward] materializes into your hands!")
+					else
+						to_chat(user, "[killreward] materializes onto the floor.")
```
Comment (deathride58) [inline modular_citadel/code/game/machinery/wishgranter.dm:89]: Could also add a 0.8 click delay multiplier for this wish.

**Comment by Michiyamenotehifunana** [comment]
Alright, that should be everything wrapped up.

File: modular_citadel/code/game/machinery/wishgranter.dm [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+			user.dna.add_mutation(TK)
+			user.next_move_modifier *= 0.5	//half the delay between attacks!
+			to_chat(user, "Things around you feel slower!")
+			charges--
+			insisting = FALSE
+			to_chat(user, "You have a very great feeling about this!")
+		else
+			to_chat(user, "The Wish Granter awaits your wish.")
+			var/wish = input("You want...","Wish") as null|anything in list("Power","Wealth","The Station To Disappear","To Kill","Nothing")
+			switch(wish)
```
Comment (silicons) [inline modular_citadel/code/game/machinery/wishgranter.dm:30]: Need to add checking for charges here or it's possible to use one multiple times even when it should have no charges left. Alternatively, add QDELETED(src) checks.

File: modular_citadel/code/game/machinery/wishgranter.dm (hunk 3 above)
Comment (silicons) [reply modular_citadel/code/game/machinery/wishgranter.dm:89]: Addressed

**Review by silicons — CHANGES_REQUESTED** [review]
Just get the QDELETED/charges checks in and we should be good.

File: modular_citadel/code/game/machinery/wishgranter.dm (hunk 4 above)
Comment (Michiyamenotehifunana) [reply modular_citadel/code/game/machinery/wishgranter.dm:30]: Would it not be caught in line 2 of the code, or is this level paranoia required?

**Review by silicons — APPROVED** [review]

**Pull request merged**